src/test/java/spoon/reflect/visitor/DefaultJavaPrettyPrinterTest.java

```diff
@@ -247,4 +257,56 @@ public Stream<? extends Arguments> provideArguments(ExtensionContext extensionCo
             );
         }
     }
+
+    @Nested
+    class SquareBracketsForArrayInitialization_ArrayIsBuiltUsingFactoryMethods {
+        @GitHubIssue(issueNumber = 4887, fixed = true)
+        @Test
+        void bracketsShouldBeAttachedToTypeByDefault() {
+            // contract: the square brackets should be attached to type by default when array is built using factory methods
+            // arrange
+            Launcher launcher = new Launcher();
+            Factory factory = launcher.getFactory();
+
+            CtArrayTypeReference<Integer> arrayTypeReference = factory.createArrayTypeReference();
+            arrayTypeReference.setComponentType(factory.Type().INTEGER_PRIMITIVE);
+            CtNewArray<Integer> newArray = factory.createNewArray();
+            newArray.setValueByRole(CtRole.TYPE, arrayTypeReference);
+            List<CtLiteral<Integer>> elements = new ArrayList<>(List.of(factory.createLiteral(3)));
+            newArray.setValueByRole(CtRole.EXPRESSION, elements);
+            CtLocalVariable<Integer> localVariable = factory.createLocalVariable(arrayTypeReference, "intArray", newArray);
+
+            // act
+            String actualStringRepresentation = localVariable.toString();
+
+            // assert
+            assertThat(actualStringRepresentation, equalTo("int[] intArray = new int[]{ 3 }"));
+        }
+
+        @Test
+        void bracketsShouldBeAttachedToIdentifierIfSpecified() {
+            // contract: the square brackets should be attached to identifier if specified explicitly
+            // arrange
+            Launcher launcher = new Launcher();
+            Factory factory = launcher.getFactory();
+
+            CtArrayTypeReference<CtElement> arrayTypeReference = factory.createArrayTypeReference();
+            CtTypeReference<CtElement> arrayType = factory.Type().createReference(CtElement.class);
+            arrayTypeReference.setComponentType(arrayType);
+            CtNewArray<CtElement> newArray = factory.createNewArray();
+            newArray.setValueByRole(CtRole.TYPE, arrayTypeReference);
+            List<CtElement> elements = new ArrayList<>(List.of(factory.createLiteral(1.0f)));
+            newArray.setValueByRole(CtRole.EXPRESSION, elements);
+            CtLocalVariable<CtElement> localVariable = factory.createLocalVariable(arrayTypeReference, "spoonElements", newArray);
+
+
+            // act
+            ((CtArrayTypeReferenceImpl<?>) arrayTypeReference).setDeclarationKind(CtArrayTypeReferenceImpl.DeclarationKind.IDENTIFIER);
```

Comment:
I think "hiding" that functionality makes sense - we probably shouldn't encourage people to make use of it. Otherwise you'd also need to add support for it on return types, i.e. `public int method()[] { return new int[10]; }` 😉

Reply:
> `public int method()[] { return new int[10]; }`

Wow. I did not know that is a valid syntax. Maybe we can add support for it if someone ever reports a bug. However, it is so unlikely people shall use this syntax which is good :) 

> Additionally, I am not sure how much value this offers as attaching the brackets to the name (or after the method...) is not a really common style from what I have gathered.

Right. It does not seem useful from a transformation perspective so we can keep it as is.